### PR TITLE
Sliced Cheese & Tomatoes

### PR DESCRIPTION
**Pull request by Nexusuxen**
## About the PR 

**Adds:**
Sliced Cheese, Sliced Green Cheese, Sliced Tomatoes, BLTs, and Weird Cheeseburgers
Sliced Cheese is made by using a cutting implement on a cheese wedge, giving 4 slices of cheese per wedge
Tomatoes will give 3 sliced tomatoes per tomato.
Weird Cheeseburgers also contain ????/"yuck", to guarantee food poisoning
BLTs come in Regular, Elvisbread, and Spooky Bread variants.

**Changes:**
Recipes for multiple foods now require cheese slices instead of cheese wedges, boosting Cheese Efficiency for chefs
Hotdogs now require 1 meatpaste instead of 2 hotdogs to make.
Meatloaf now requires meatpaste, 1 eggs, and a loaf of bread (haha get it, meat LOAF. ha.)

**Fixes:**
Inability to make regular cheeseburgers with monkey meat, and inability to include cheese on custom sandwiches, as both Monkey Cheeseburgers and Cheese Sandwiches now require 2 slices of cheese (regular Cheeseburgers only require 1 slice).

Prior to this, trying to put cheese into a custom sandwich would turn it into a cheese sandwich even if you had other ingredients. Similarly, using monkeymeat in the cheeseburger recipe would result in making a monkey cheeseburger (for reference, the monkey cheeseburger has special potentially unwanted effects when eaten).

A couple really trivial fixes, such as a typo or two and adding some reagents to some related foods for a little more consistency.

**Tweaks:**
Makes slicing of Pepperoni, Cheese, Weird Cheese, and Tomatoes occur in an attackby() proc inherent to `/item/reagent_containers/food`. New variables also inherent to `/food` are: `sliceable` (boolean), `slice_product` (obj path), and `slice_amount` (int). I'm too tired/inexperienced for this, but a system that pokes the src obj (the one being sliced) and giving it the option to, if there's a proc someone made for the src obj, do special things upon slicing like inheriting ingredients or setting specific reagents, etc., might allow for this to become more universal with special things like slicing pizza.

## Why's this needed? 

Cheese is honestly annoyingly hard to come by and is very limited in quantity, with the chef starting with just 7 wedges. Each taco, each cheeseburger, each lasagna, etc. needs one out of the 7 precious wedges of cheese you have. You can always make more but it's a messy process and trying to store 50 wedges of cheeses is initially funny but ends up being clunky and annoying.

The slicing of food being inherent to `/food` is intended to (hopefully) simplify the addition of new sliced ingredients, as it's now just 3 easily adjustable variables that you need to add to an item instead of like 7 or 8 lines of code. Unfortunately not applicable to all food items as some will have custom text or effects or procs that run when the item is cut (e.g pizza)

Also some recipe tweaks to make chefs lives a little nicer (seriously why did hotdogs take the equivalent of 2 pepperoni logs of meat)

## Extra Info

Sliced Cheese sprites (regular and green) by Erinexx

**New Recipes:**
_Weird Cheeseburger:_
1x Dough, 1x Meat, 1x Weird Sliced Cheese

_BLT (Normal, Elvis, and Haunted):_
2x Bread (Normal, Elvis, or Dread), 1x Raw Bacon, 1x Sliced Tomato, 1x Lettuce Leaf

**Changed recipes that now require cheese slices:**
_Cheeseburger (x1)
Monkey Cheeseburger (x2)
Cheeseborger (x1)
Coronator (x2)
Lasagna (x2)
Cheese Sandwich (all varieties) (x2)
Meatball Sub (x1)
Cheese Toast (regular & elvis) (x1)
Cheesy Garlic Bread (x1)
MacGuffin (x1)
Omelette (x1)
Tacos (x1)_

**Generally changed recipes**
_Meatloaf:_
1x Meatpaste, 1x Egg, 1x Bread Loaf
_Hotdog:_
1x Meatpaste

## Pictures

image
Top row, left to right: Haunted BLT, Elviswich BLT, regular BLT, Weird Cheeseburger, Sliced Green Cheese, Sliced Cheese, Sliced Tomato
Everything else are existing foods whose recipes have been adjusted to use the new cheese slices. See the earlier list for more details.

## Changelog

```changelog
(u)Nexusuxen
(+) Adds sliced cheese & tomatoes, and a couple recipe fixes. 1 cheese wedge = 4 slices, 1 tomato = 3 slices. Cheese slice sprites by Erinexx.
(+) Multiple recipes now use cheese slices instead of whole wedges. BLTs and Weird Cheeseburgers added. Minor recipe tweaks.
```

### CODE REVIEW

File: code/modules/food_and_drink/sandwiches.dm [hunk 1]
```diff
@@ -137,6 +161,14 @@
 		food_effects = list("food_hp_up_big")
 		meal_time_flags = MEAL_TIME_FORBIDDEN_TREAT
 
+	spooky_blt
+		name = "haunted BLT"
```
Comment (genessee-forgot-his-password-again) [inline code/modules/food_and_drink/sandwiches.dm:165]:
```suggestion
		name = "Boo-LT"
```

Comment (genessee-forgot-his-password-again) [reply code/modules/food_and_drink/sandwiches.dm:165]: it's punny!

Comment (Nexusuxen) [reply code/modules/food_and_drink/sandwiches.dm]: 10,000% yes, I couldn't think of a good name at the time anyways lmao

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 2]
```diff
@@ -17,6 +17,9 @@
 	var/from_emagged_oven = 0 // to prevent re-rolling of food in emagged ovens
 	var/doants = 1
 	var/made_ants = 0
+	var/sliceable = FALSE // for if a food is intended to be able to be sliced
+	var/slice_product = null // what product to spawn when sliced
+	var/slice_amount = 0 // how much product to spawn when sliced
```
Comment (ZeWaka) [inline code/modules/chemistry/tools/food_and_drink.dm:22]: doc comments

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drink/ingredients.dm [hunk 3]
```diff
@@ -319,6 +319,11 @@
 	heal_amt = 1
 	food_color = "#FFD700"
 	custom_food = 1
+	initial_volume = 5
+	initial_reagents = "cheese"
+	sliceable = 1
+	slice_product = /obj/item/reagent_containers/food/snacks/ingredient/cheeseslice
```
[the remark this answers is not in the document]
Comment (ZeWaka) [reply code/modules/food_and_drink/ingredients.dm:327]: I am unsure why this and related comments was resolved when it actually wasn't done.

**Pull request merged**